libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Twilio.Exceptions;
+using Twilio.Rest.Api.V2010.Account;
+using Twilio.Security;
+
+namespace Microsoft.Bot.Builder.Adapters.Twilio
+{
+    public static class TwilioHelper
+    {
+        /// <summary>
+        /// Extracts attachments (if any) from a twilio message and returns them in an Attachments array.
+        /// </summary>
+        /// <param name="message">The TwilioEvent message.</param>
+        /// <returns>An Attachments array with the converted attachments.</returns>
+        public static List<Attachment> GetMessageAttachments(TwilioEvent message)
+        {
+            var attachments = new List<Attachment>();
+            if (int.TryParse(message.NumMedia, out var numMediaResult) && numMediaResult > 0)
+            {
+                for (var i = 0; i < numMediaResult; i++)
+                {
+                    var attachment = new Attachment()
+                    {
+                        ContentType = message.MediaContentTypes[i],
+                        ContentUrl = message.MediaUrls[i].AbsolutePath,
+                    };
+                    attachments.Add(attachment);
+                }
+            }
+
+            return attachments;
+        }
+
+        /// <summary>
+        /// Converts a query string to a dictionary with key-value pairs.
+        /// </summary>
+        /// <param name="query">The query string to convert.</param>
+        /// <returns>A dictionary with the query values.</returns>
+        public static Dictionary<string, string> QueryStringToDictionary(string query)
+        {
```

Comment:
If _query_ is _**null**_, should we return an _empty Dictionary_? Or should we return _null_?

Reply:
I lean in favor of returning null